src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs

```diff
@@ -13019,6 +13019,105 @@ private static Vector512<T> IsNegative<T>(Vector512<T> vector)
             return Vector512.LessThan(vector, Vector512<T>.Zero);
         }
 
+        /// <summary>poly = C0 + C1*r^3 + C2*r^5 + C3*r^7 + C4*r^9</summary>
+        private static Vector128<T> PolyEvalOdd9<T>(Vector128<T> r, Vector128<T> c0, Vector128<T> c1, Vector128<T> c2, Vector128<T> c3, Vector128<T> c4)
+        {
+            Vector128<T> r2 = r * r;
+            Vector128<T> r4 = r2 * r2;
+
+            Vector128<T> a0 = c2 * r2 + c1;
+            Vector128<T> a1 = a0 * r2 + c0;
+            Vector128<T> a2 = (c3 * r2 + c4 * r4) * r4;
+
+            return r * (a1 + a2);
+        }
+
+        /// <summary>poly = C0 + C1*r^3 + C2*r^5 + C3*r^7 + C4*r^9</summary>
+        private static Vector256<T> PolyEvalOdd9<T>(Vector256<T> r, Vector256<T> c0, Vector256<T> c1, Vector256<T> c2, Vector256<T> c3, Vector256<T> c4)
+        {
+            Vector256<T> r2 = r * r;
+            Vector256<T> r4 = r2 * r2;
+
+            Vector256<T> a0 = c2 * r2 + c1;
+            Vector256<T> a1 = a0 * r2 + c0;
+            Vector256<T> a2 = (c3 * r2 + c4 * r4) * r4;
+
+            return r * (a1 + a2);
+        }
+
+        /// <summary>poly = C0 + C1*r^3 + C2*r^5 + C3*r^7 + C4*r^9</summary>
+        private static Vector512<T> PolyEvalOdd9<T>(Vector512<T> r, Vector512<T> c0, Vector512<T> c1, Vector512<T> c2, Vector512<T> c3, Vector512<T> c4)
+        {
+            Vector512<T> r2 = r * r;
+            Vector512<T> r4 = r2 * r2;
+
+            Vector512<T> a0 = c2 * r2 + c1;
+            Vector512<T> a1 = a0 * r2 + c0;
+            Vector512<T> a2 = (c3 * r2 + c4 * r4) * r4;
+
+            return r * (a1 + a2);
+        }
+
+        /// <summary>poly = C1*r^3 + C2*r^5 + C3*r^7 + C4*r^9 + C5*r^11 + C6*r^13 + C7*r^15 + C8*r^17</summary>
+        private static Vector128<T> PolyEvalOdd17<T>(Vector128<T> r, Vector128<T> c1, Vector128<T> c3, Vector128<T> c5, Vector128<T> c7, Vector128<T> c9, Vector128<T> c11, Vector128<T> c13, Vector128<T> c15)
+        {
+            Vector128<T> r2 = r * r;
+            Vector128<T> r4 = r2 * r2;
+            Vector128<T> r6 = r4 * r2;
+            Vector128<T> r10 = r6 * r4;
+            Vector128<T> r14 = r10 * r4;
+
+            Vector128<T> a1 = c1 + c3 * r2;
+            Vector128<T> a2 = c5 + c7 * r2;
+            Vector128<T> a3 = c9 + c11 * r2;
+            Vector128<T> a4 = c13 + c15 * r2;
+
+            Vector128<T> b1 = a1 * r2 + a2 * r6;
+            Vector128<T> b2 = r10 * a3 + r14 * a4;
```

Comment:
Is the change in order intentional here?
```suggestion
            Vector128<T> b2 = a3 * r10 + a4 * r14;
```

Reply:
It's a direct port of the amd C code. I tried to keep it as close as possible: the rhs of the expressions here are all just copy/paste. But I can revisit it.